src/dashboard/tests/test_helpers.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import __builtin__
+import json
+import os
+import requests
+import tempfile
+
+from amclient import AMClient
+from components import helpers
+
+
+def test_stream_mets_from_disconnected_storage(mocker):
+    custom_error_message = "Error connecting to the storage service"
+    mocker.patch.object(AMClient, "__init__", return_value=None)
+    mocker.patch.object(
+        helpers, "get_setting", return_value="http://storage-service-url/"
+    )
+    mocker.patch.object(
+        AMClient,
+        "extract_file",
+        side_effect=requests.exceptions.ConnectionError(custom_error_message),
+    )
+    response = helpers.stream_mets_from_storage_service(
+        transfer_name="mets_transfer", sip_uuid="11111111-1111-1111-1111-111111111111"
+    )
+    err = json.loads(response.content)
+    assert err.get("success") is False
+    assert custom_error_message in err.get("message")
+    assert response.status_code == 503
+
+
+def test_stream_mets_from_storage_no_file(mocker, tmp_path):
+    custom_error_message = "METS file cannot be opened"
+    mocker.patch.object(AMClient, "__init__", return_value=None)
+    mocker.patch.object(
+        helpers, "get_setting", return_value="http://storage-service-url/"
+    )
+    mocker.patch.object(AMClient, "extract_file")
+    mocker.patch.object(__builtin__, "open", side_effect=IOError(custom_error_message))
+    response = helpers.stream_mets_from_storage_service(
+        transfer_name="mets_transfer", sip_uuid="22222222-2222-2222-2222-222222222222"
+    )
+    err = json.loads(response.content)
+    assert err.get("success") is False
+    assert custom_error_message in err.get("message")
+    assert response.status_code == 404
+
+
+def test_stream_mets_from_storage_success(mocker):
+    METS = """<?xml version='1.0' encoding='UTF-8'?>
+    <mets:mets xmlns:mets="http://www.loc.gov/METS/" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/METS/ http://www.loc.gov/standards/mets/version1121/mets.xsd">
+        <mets:metsHdr CREATEDATE="2020-01-20T15:22:15"/>
+    </mets:mets>
+    """
+    test_tmp_dir = "{}{}".format(os.path.sep, os.path.join("test-tmp-dir", ""))
+    sip_uuid = "33333333-3333-3333-3333-333333333333"
+    mets_file = "METS.{}.xml".format(sip_uuid)
+    mets_file_path = "{}{}".format(test_tmp_dir, mets_file)
```

Comment:
Is this whole path setup necessary? We could use any name (`anything.pdf`) and we just care `open` was called with it, right? I guess it's a matter of taste how "real world" you want to make the test looks like, but we're mocking anyway :man_shrugging:

Reply:
I'd like to simplify this but ultimately keep this as-is too, my rationale is that our tests can reveal a lot more about the data that we're passing through the system than perhaps is revealed in any of our comments/docs etc. Keeping the tests real for me is a way to show truthfully what we expect and open up opportunities for other devs to consider what to do with that.

Reply:
Hmm, in the end I didn't manage to change that much at all, but other parts of the tests are simplified I think/made less redundant. More in the re-review comment!